scipy/stats/_quantile.py

```diff
@@ -335,3 +345,25 @@ def _quantile_hd(y, p, n, xp):
     w = xpx.at(w, xp.isnan(w)).set(0)
     res = xp.vecdot(w, y, axis=-1)
     return xp.moveaxis(res, 0, -1)
+
+
+def _quantile_winsor(y, p, n, method, xp):
+    ops = dict(winsor_less=(xp.floor, xp.ceil),
+               winsor_more=(xp.ceil, xp.floor),
+               winsor_round=(xp.round, xp.round))
+    op_left, op_right = ops[method]
+    j = xp.where(p < 0.5, op_left(p*n), op_right(n*p - 1))
+    return xp.take_along_axis(y, xp.astype(j, xp.int64), axis=-1)
+
+
+def _quantile_trim(y, p, n, method, xp):
+    ops = dict(trim_less=(xp.floor, xp.ceil),
+               trim_more=(xp.ceil, xp.floor),
+               trim_round=(xp.round, xp.round))
+    op_left, op_right = ops[method]
+    j = xp.where(p < 0.5, op_left(p*n)-1, op_right(n*p))
+    j_clip = xp.clip(j, 0*n, n-1)
+    res = xp.take_along_axis(y, xp.astype(j_clip, xp.int64), axis=-1)
+    res[j < 0] -= 1
+    res[j >= n] += 1
```

Comment:
Whether this is needed depends on what options we end up with and whether we standardize the use of strict vs non-strict inequalities in masking.... easier to discuss in real time.